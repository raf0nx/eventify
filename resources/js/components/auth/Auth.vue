<template>
	<v-main>
		<v-container class="fill-height py-0" fluid>
			<v-row class="fill-height">
				<v-col cols="12" md="7" class="pa-0 auth-panel">
					<div
						class="auth-panel__container pa-4 d-flex flex-column justify-space-between"
					>
						<div class="d-flex logo">
							<v-icon class="white--text mr-1">mdi-buffer</v-icon>
							<span class="logo__text white--text font-weight-bold"
								>Eventify</span
							>
						</div>
						<div class="d-flex flex-column align-center mb-16">
							<h2 class="auth-panel__title white--text">
								Welcome back!
							</h2>
							<span
								class="auth-panel__subtitle grey--text text--lighten-3 mt-4"
								>Sign In to continue using my application!</span
							>
						</div>
						<div class="d-flex justify-center">
							<span class="white--text font-weight-bold"
								>v. 0.0</span
							>
						</div>
					</div>
				</v-col>
				<transition name="scale" mode="out-in">
					<router-view />
				</transition>
			</v-row>
		</v-container>
	</v-main>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	@Component
	export default class Auth extends Vue {}
</script>

<style>
	.auth-panel {
		background-image: url("https://i.pinimg.com/originals/c4/6f/3c/c46f3cfbff1b198e0276e4d577a4464c.png");
		background-position: center;
		background-repeat: no-repeat;
		background-size: cover;
	}

	.auth-panel__container {
		height: 100%;
	}

	.auth-panel__title {
		font-size: 4rem;
	}

	.auth-panel__subtitle {
		font-size: 1.5rem;
	}

	.logo__text {
		font-size: 1.5rem;
	}

	.auth-form__title {
		font-size: 3rem;
	}
</style>