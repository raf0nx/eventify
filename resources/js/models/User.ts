export interface User {
    id: Number;
    name: String;
    email: string;
    created_at: Date;
    email_verified_at?: Date;
    updated_at: Date;
}
