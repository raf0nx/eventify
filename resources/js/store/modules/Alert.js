export const namespaced = true;

export const state = {
    message: null,
    showAlert: false
};

export const mutations = {
    SET_ALERT(state, message) {
        state.showAlert = true;
        state.message = message;
    }
};

export const actions = {};

export const getters = {
    message: state => {
        return state.message;
    },
    showAlert: state => {
        return state.showAlert;
    }
};
