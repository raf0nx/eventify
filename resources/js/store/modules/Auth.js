import router from "../../router/router";
import AuthService from "../../services/AuthService";

export const namespaced = true;

export const state = {
    user: null,
};

export const mutations = {
    SET_USER(state, user) {
        state.user = user;
    },
};

export const actions = {
    async getAuthUser({ commit }) {
        try {
            const authUser = await AuthService.getAuthUser();
            commit("SET_USER", authUser.data);
            return authUser.data;
        } catch (error) {
            commit("SET_USER", null);
        }
    },
    logout({ commit }) {
        try {
            AuthService.logout();
            commit("SET_USER", null);
            if (router.currentRoute.name !== "Login") {
                router.push({ name: "Login" });
            }
        } catch (error) {
            console.log(error);
        }
    }
};

export const getters = {
    authUser: state => {
        return state.user;
    },
};
