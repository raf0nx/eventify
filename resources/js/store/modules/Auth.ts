import {
    Module,
    VuexModule,
    Mutation,
    Action,
    getModule
} from "vuex-module-decorators";

import store from "@/store/store.ts";
import router from "@/routes/router.ts";
import * as AuthService from "@/services/AuthService.ts";

export interface AuthState {
    user: any;
}

@Module({ dynamic: true, store, name: "auth" })
class Auth extends VuexModule implements AuthState {
    public user: any = null;

    @Mutation
    private SET_USER(userPayload: any) {
        this.user = userPayload;
    }
    @Action
    public async getAuthUser() {
        try {
            const authUser = await AuthService.getAuthUser();
            this.SET_USER(authUser.data);
            return authUser.data;
        } catch (error) {
            this.SET_USER(null);
        }
    }

    @Action
    public async logout() {
        try {
            await AuthService.logout();
            this.SET_USER(null);
            if (router.currentRoute.name !== "Login") {
                router.push({ name: "Login" });
            }
        } catch (error) {
            console.log(error);
        }
    }

    public get authUser() {
        return this.user;
    }
}

export const AuthModule = getModule(Auth);
