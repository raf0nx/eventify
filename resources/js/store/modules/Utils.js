export const namespaced = true;

export const state = {
    showSnackbar: false,
    showAlert: false,
    loading: false
};
export const mutations = {
    SET_ALERT(state, setAlert) {
        state.showAlert = setAlert;
    },
    SET_LOADING(state, isLoading) {
        state.loading = isLoading;
    },
    SET_SNACKBAR(state, setSnackbar) {
        state.showSnackbar = setSnackbar;
    }
};
export const actions = {
    setSnackbar({ commit }) {
        commit("SET_SNACKBAR", true);
        setTimeout(() => commit("SET_SNACKBAR", false), 2500);
    }
};

export const getters = {
    showAlert: state => {
        return state.showAlert;
    },
    isLoading: state => {
        return state.loading;
    },
    showSnackbar: state => {
        return state.showSnackbar;
    }
};
