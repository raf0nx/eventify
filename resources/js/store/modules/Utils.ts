import {
    Module,
    VuexModule,
    Mutation,
    Action,
    getModule
} from "vuex-module-decorators";

import store from "@/store/store";
import { SnackbarModel } from "@/models/Snackbar";
import { AlertModel } from "@/models/Alert";

export interface UtilsState {
    snackbarNotification: SnackbarModel;
    alertNotification: AlertModel;
    isLoading: boolean;
    navDrawer: boolean;
}

@Module({ dynamic: true, store, name: "utils" })
class Utils extends VuexModule implements UtilsState {
    snackbarNotification = new SnackbarModel();
    alertNotification = new AlertModel();
    isLoading = false;
    navDrawer = false;

    @Mutation
    private SET_ALERT(alert: AlertModel): void {
        this.alertNotification = alert;
    }

    @Mutation
    private SET_LOADER(isLoading: boolean): void {
        this.isLoading = isLoading;
    }

    @Mutation
    private SET_SNACKBAR(snackbar: SnackbarModel): void {
        this.snackbarNotification = snackbar;
    }

    @Mutation
    private SET_NAV_DRAWER(setNavDrawer: boolean): void {
        this.navDrawer = setNavDrawer;
    }

    @Action
    setAlert(alert: AlertModel): void {
        this.SET_ALERT(alert);
    }

    @Action
    setLoader(isLoading: boolean): void {
        this.SET_LOADER(isLoading);
    }

    @Action
    setSnackbar(snackbar: SnackbarModel): void {
        this.SET_SNACKBAR(snackbar);
    }

    @Action
    setNavDrawer(setNavDrawer: boolean): void {
        this.SET_NAV_DRAWER(setNavDrawer);
    }

    get alert(): AlertModel {
        return this.alertNotification;
    }

    get loader(): boolean {
        return this.isLoading;
    }

    get snackbar(): SnackbarModel {
        return this.snackbarNotification;
    }

    get isNavDrawer(): boolean {
        return this.navDrawer;
    }
}

export const UtilsModule = getModule(Utils);
