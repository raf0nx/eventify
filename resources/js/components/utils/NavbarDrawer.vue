<template>
	<v-navigation-drawer v-model="navDrawer" app temporary bottom>
		<template v-slot:prepend>
			<v-list-item two-line>
				<v-badge
					bordered
					bottom
					color="green accent-4"
					dot
					offset-x="26"
					offset-y="24"
				>
					<v-list-item-avatar>
						<img
							src="https://randomuser.me/api/portraits/women/81.jpg"
						/>
					</v-list-item-avatar>
				</v-badge>
				<v-list-item-content>
					<v-list-item-title>{{ username }}</v-list-item-title>
					<v-list-item-subtitle>Active</v-list-item-subtitle>
				</v-list-item-content>
			</v-list-item>
		</template>

		<v-divider></v-divider>

		<v-list nav dense>
			<v-list-item-group active-class="deep-purple--text text--darken-4">
				<v-list-item :to="{ name: 'Dashboard' }">
					<v-list-item-icon>
						<v-icon>mdi-view-dashboard</v-icon>
					</v-list-item-icon>
					<v-list-item-content>Dashboard</v-list-item-content>
				</v-list-item>

				<v-list-item>
					<v-list-item-icon>
						<v-icon>mdi-account</v-icon>
					</v-list-item-icon>
					<v-list-item-content>Profile</v-list-item-content>
				</v-list-item>

				<v-list-item>
					<v-list-item-icon>
						<v-icon>mdi-heart</v-icon>
					</v-list-item-icon>
					<v-list-item-content>Favourites</v-list-item-content>
				</v-list-item>

				<v-list-item>
					<v-list-item-icon>
						<v-icon>mdi-cogs</v-icon>
					</v-list-item-icon>
					<v-list-item-content>Settings</v-list-item-content>
				</v-list-item>
			</v-list-item-group>
		</v-list>

		<template v-slot:append>
			<v-btn
				color="deep-purple"
				class="white--text font-weight-bold mt-12"
				block
				tile
				@click="logout()"
				data-cy="logout_button"
			>
				<v-icon class="mr-1">mdi-logout</v-icon>
				Logout
			</v-btn>
		</template>
	</v-navigation-drawer>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { AuthModule } from "@modules/Auth";

	@Component
	export default class NavbarDrawer extends Vue {
		get navDrawer(): boolean {
			return UtilsModule.isNavDrawer;
		}

		set navDrawer(value: boolean) {
			UtilsModule.setNavDrawer(value);
		}

		get username(): string {
			return AuthModule.authUser ? AuthModule.authUser.name : "unidentified";
		}

		logout(): void {
			AuthModule.logout();
		}
	}
</script>

<style>
</style>