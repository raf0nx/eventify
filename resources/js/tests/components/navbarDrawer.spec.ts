import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import axios from "axios";

import NavbarDrawer from "@/components/utils/NavbarDrawer.vue";
import { AuthModule } from "@/store/modules/Auth";
import { UtilsModule } from "@/store/modules/Utils";
import { user } from "@/tests/constans/User";

jest.mock("axios");

describe("NavbarDrawer.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<NavbarDrawer>;
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
        wrapper = shallowMount(NavbarDrawer, { localVue, vuetify });
    });

    afterEach(() => {
        wrapper.destroy();
        jest.clearAllMocks();
    });

    it("Should match snapshot", () => {
        // Assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should get user", async () => {
        // Arrange
        // @ts-ignore
        axios.get.mockImplementationOnce(() => Promise.resolve({ data: user }));

        // Act
        const authUser = await AuthModule.getAuthUser();

        // Assert
        // @ts-ignore
        expect(wrapper.vm.user).toEqual(authUser);
    });

    it("Should not get user", async () => {
        // Arrange
        // @ts-ignore
        axios.get.mockImplementationOnce(() => Promise.reject());

        // Act
        await AuthModule.getAuthUser();

        // Assert
        // @ts-ignore
        expect(wrapper.vm.user).toBeNull();
    });

    it("Should set navbar drawer", () => {
        // Act
        // @ts-ignore
        wrapper.vm.navDrawer = true;

        // Assert
        expect(UtilsModule.isNavDrawer).toBe(true);
    });

    it("Should logout user", async () => {
        // Arrange
        // @ts-ignore
        axios.post.mockResolvedValue();

        // Act
        // @ts-ignore
        await wrapper.vm.logout();

        // Assert
        expect(AuthModule.authUser).toBeNull();
    });
});
