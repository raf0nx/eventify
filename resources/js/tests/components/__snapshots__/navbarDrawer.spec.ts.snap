// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarDrawer.vue Should match snapshot 1`] = `
"<v-navigation-drawer-stub app=\\"true\\" mobilebreakpoint=\\"1264\\" bottom=\\"true\\" height=\\"100vh\\" minivariantwidth=\\"56\\" src=\\"\\" tag=\\"nav\\" temporary=\\"true\\" width=\\"256\\">
  <v-divider-stub></v-divider-stub>
  <v-list-stub tag=\\"div\\" dense=\\"true\\" nav=\\"true\\">
    <v-list-item-group-stub activeclass=\\"deep-purple--text text--darken-4\\" tag=\\"div\\">
      <v-list-item-stub activeclass=\\"\\" to=\\"[object Object]\\" tag=\\"div\\">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-view-dashboard</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>Dashboard</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-account</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>Profile</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-heart</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>Favourites</v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-cogs</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>Settings</v-list-item-content-stub>
      </v-list-item-stub>
    </v-list-item-group-stub>
  </v-list-stub>
</v-navigation-drawer-stub>"
`;
